### PR TITLE
Add HTML5 dropdown for image selector with cpu,gpu defaults

### LINKED ISSUE #309: JupyterHub Spawner should have a UI element that specifies default docker images
The JupyterHub Spawner provides a UI in which users can specify the docker image to use for the notebooks.

It would be great if we could provide a drop down list that was prepopulated with the default images provided by Kubeflow.

It would also be nice if this list could be customizable via ksonnet so users could add their own images.

According to jupyterhub/jupyterhub#1580 we should be able to do this just by adding some properties to our KubeSpawner.

For reference.
https://github.com/jupyterhub/kubespawner

### PR DESCRIPTION
**Pull request by pdmack**
Fixes #309 
More work/investigation required of best way to parameterize the images for the UI list, for now hardcoded

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/402)

### CODE REVIEW

**Comment by jlewi** [comment]
This is fantastic!

File: kubeflow/core/jupyterhub_spawner.py [hunk 1]
```diff
@@ -5,10 +5,16 @@
...
 class KubeFormSpawner(KubeSpawner):
+
+  # relies on HTML5 for image datalist
   def _options_form_default(self):
     return '''
     <label for='image'>Image</label>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
-    <input name='image' placeholder='repo/image:tag'></input>
+    <input list="image" name="image" placeholder='repo/image:tag'>
+    <datalist id="image">
+      <option value="gcr.io/kubeflow/tensorflow-notebook-cpu">
```
Comment (jlewi) [inline kubeflow/core/jupyterhub_spawner.py]: The latest images are in
gcr.io/kubeflow-images-staging/tensorflow-notebook-cpu
gcr.io/kubeflow-images-staging/tensorflow-notebook-gpu

Its part of our evolving release process.

File: testing/workflows/vendor/kubeflow/core/jupyterhub.libsonnet [hunk 2]
```diff
@@ -19,7 +19,11 @@ class KubeFormSpawner(KubeSpawner):
   def _options_form_default(self):
     return '''
     <label for='image'>Image</label>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
-    <input name='image' placeholder='repo/image:tag'></input>
+    <input list="image" name="image" placeholder='repo/image:tag'>
+    <datalist id="image">
+      <option value="gcr.io/kubeflow/tensorflow-notebook-cpu">
+      <option value="gcr.io/kubeflow/tensorflow-notebook-gpu">
```
Comment (jlewi) [inline testing/workflows/vendor/kubeflow/core/jupyterhub.libsonnet]: gcr.io/kubeflow-images-staging/tensorflow-notebook-cpu
gcr.io/kubeflow-images-staging/tensorflow-notebook-gpu

**Comment by jlewi** [comment]
Can you also update the docs here
https://github.com/kubeflow/kubeflow/blob/master/user_guide.md#bringing-up-a-jupyter-notebook

Change the images
gcr.io/kubeflow/tensorflow-notebook-cpu:v1 
gcr.io/kubeflow/tensorflow-notebook-gpu:8fbc341245695e482848ac3c2034a99f7c1e576

to 
gcr.io/kubeflow-images-staging/tensorflow-notebook-cpu
gcr.io/kubeflow-images-staging/tensorflow-notebook-gpu

Or maybe just delete them altogether since the UI is self explanatory.

**Comment by pdmack** [comment]
Updated per @jlewi request

**Comment by pdmack** [comment]
screen shot 2018-03-10 at 9 26 57 am

**Comment by pdmack** [comment]
screen shot 2018-03-10 at 9 30 11 am

**Pull request merged**